server/client_manager.py

```diff
@@ -288,7 +288,14 @@ def get_targets_by_ip(self, ip):
             if client.get_ip() == ip:
                 clients.append(client)
         return clients
-		
+
+    def get_targets_by_hdid(self, hdid):
+        clients = []
+        for client in self.clients:
+            if client.get_hdid() == hdid:
+                clients.append(client)
+        return clients
```

Comment:
you can use a generator/`yield` here but I get that you want to be consistent with the code.